### PR TITLE
Deprecate not configuring explicitly a provider for custom_authenticators when there is more than one registered provider

### LINKED ISSUE #16041: [SecurityBundle] Deprecate not configuring explicitly a provider for cu…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#43835
| PR author(s) | @lyrixx
| Merged in    | 5.4

### PR DESCRIPTION
**Pull request by mdoutreluingne**
Fixes #16041

### CODE REVIEW

File: security/custom_authenticator.rst [hunk 1]
```diff
@@ -130,6 +130,12 @@ The authenticator can be enabled using the ``custom_authenticators`` setting:
             ;
         };
 
+.. deprecated:: 5.4
+
+    Having multiple user providers and a ``custom_authenticator`` and not having specified the user provider 
+    to use has been deprecated in Symfony 5.4. If you have registered multiple user providers, you must set the 
+    ``provider`` key to one of the configured providers, even if your custom authenticators don't use it.
```
Comment (xabbuh) [inline security/custom_authenticator.rst]: I think I would reword this a bit:

>  If you have registered multiple user providers, you must set the 
    ``provider`` key to one of the configured providers, even if your custom authenticators don't use it. Not doing so is deprecated in Symfony 5.4.

Comment (mdoutreluingne) [reply security/custom_authenticator.rst]: I agree, it's done

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
Thank you @mdoutreluingne.
